visvis2/geometry/_dynmesh.py

```diff
@@ -0,0 +1,11 @@
+from ._base import Geometry
+
+
+# I think this is where this thing goes, right?
+
+
+class DynamicMeshGeometry(Geometry):
+    """ A datasource, euh, I mean Geometry, specifically for
+    triangulated meshes, supporting resampling and a variety of methods
+    to query the mesh and make changes to it.
+    """
```

Comment:
I would strongly recommend against trying to support this functionality through the class hierarchy.

Reply:
What would be the benefit of using a class for this anyway?

Reply:
I'm trying to see how pieces like this fit into the puzzle. So how do you think that this functionality should be fit into (the eventual) visvis2?

Reply:
Let me think about this one

Reply:
Read the section "BufferGeometry" on this page: https://threejs.org/docs/#manual/en/introduction/How-to-update-things

The linked [fiddle](http://jsfiddle.net/w67tzfhx/) showcases the approach.

Reply:
It looks like `needsUpdate` re-uploads the whole buffer, which would be *way* to slow for our purpose. But from the point of updating, things are very different anyway, because we can use mapped data.

However, we still need to perform all that state management, and you want to wrap that up in a class. So the question is whether that class should be a subclass of geometry (which would be my guess) or something else.

This is not that important right now, but I suppose it helps us to think about these things  in the same way :)

Reply:
needsUpdate is something you should use when you actually _want_ the whole buffer to be reuploaded, for example in the event of resizing the buffer. otherwise you dont need it

Reply:
> otherwise you dont need it

For uniforms this is the case. But AFAIK if you change the array data in a BufferGeometry you need to set that flag to make it sync to the GPU.

Reply:
try commenting the line in the fiddle; it still works

Reply:
I think its solved by setting [https://threejs.org/docs/#api/en/core/BufferAttribute.usage](BufferAttribute.usage) to DynamicDraw